test/module/irohad/ametsuchi/ametsuchi_test.cpp

```diff
@@ -449,74 +375,36 @@ TEST_F(AmetsuchiTest, queryGetAccountAssetTransactionsTest) {
   block.hash = block3hash;
   block.txs_number = static_cast<uint16_t>(block.transactions.size());
 
-  {
-    auto ms = storage->createMutableStorage();
-    ms->apply(block, [](const auto &, auto &, const auto &) { return true; });
-    storage->commit(std::move(ms));
-  }
+  apply(storage)(block);
 
-  {
-    auto asset1 = wsv->getAccountAsset(user2id, asset2id);
-    ASSERT_TRUE(asset1);
-    ASSERT_EQ(asset1->account_id, user2id);
-    ASSERT_EQ(asset1->asset_id, asset2id);
-    ASSERT_EQ(asset1->balance, iroha::Amount(90, 2));
-
-    auto asset2 = wsv->getAccountAsset(user3id, asset2id);
-    ASSERT_TRUE(asset2);
-    ASSERT_EQ(asset2->account_id, user3id);
-    ASSERT_EQ(asset2->asset_id, asset2id);
-    ASSERT_EQ(asset2->balance, iroha::Amount(150, 2));
-
-    auto asset3 = wsv->getAccountAsset(user1id, asset2id);
-    ASSERT_TRUE(asset3);
-    ASSERT_EQ(asset3->account_id, user1id);
-    ASSERT_EQ(asset3->asset_id, asset2id);
-    ASSERT_EQ(asset3->balance, iroha::Amount(10, 2));
-  }
+  boost::for_each(
+      combine(
+          {user2id, user3id, user1id},
+          {asset2id, asset2id, asset2id},
+          {iroha::Amount(90, 2), iroha::Amount(150, 2), iroha::Amount(10, 2)}),
+      validateAccountAsset(wsv));
 
   // Block store tests
-  blocks->getBlocks(1, 3).subscribe(
-      [block1hash, block2hash, block3hash](auto eachBlock) {
-        if (eachBlock.height == 1) {
-          EXPECT_EQ(eachBlock.hash, block1hash);
-        } else if (eachBlock.height == 2) {
-          EXPECT_EQ(eachBlock.hash, block2hash);
-        } else if (eachBlock.height == 3) {
-          EXPECT_EQ(eachBlock.hash, block3hash);
-        }
-      });
-
-  blocks->getAccountTransactions(admin).subscribe(
-      [](auto tx) { EXPECT_EQ(tx.commands.size(), 9); });
-  blocks->getAccountTransactions(user1id).subscribe(
-      [](auto tx) { EXPECT_EQ(tx.commands.size(), 1); });
-  blocks->getAccountTransactions(user2id).subscribe(
-      [](auto tx) { EXPECT_EQ(tx.commands.size(), 2); });
-  blocks->getAccountTransactions(user3id).subscribe(
-      [](auto tx) { EXPECT_EQ(tx.commands.size(), 0); });
+  validateCalls(blocks->getBlocks(1, 3),
+                [ x = 0, hashes = {block1hash, block2hash, block3hash} ](
+                    auto eachBlock) mutable {
+                  EXPECT_EQ(*(hashes.begin() + x++), eachBlock.hash);
+                },
+                3);
+
+  boost::for_each(
+      combine({admin, user1id, user2id, user3id}, {1, 1, 1, 0}, {9, 1, 2, 0}),
```

Comment:
Why do we need to combine, if we may write tuples of target data: `{admin, 1, 9}...`. This is more convenient and better to read way, isn't it?

Reply:
The variables in intializer lists are of known types (string and int), and list type is known without explicit specification. Transposing these lists to tuples will require writing tuple type explicitly somewhere, which will make code more complex.

Reply:
But it this case it will be clear to a reader.